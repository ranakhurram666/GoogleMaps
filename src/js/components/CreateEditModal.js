import React from "react";
import Modal from "react-modal";
import { connect } from "react-redux";

import { saveMarkers, displayCreateEditModal, updateMarkers } from "../actions";
import { toast } from "react-toastify";
@connect(store => {
  return {};
})
export default class CreateEditModal extends React.Component {
  constructor() {
    super();
    this.state = {
      id: null,
      name: "",
      lat: null,
      long: null
    };
  }

  componentWillReceiveProps(newProps) {
    if (
      newProps.modalInfo.markerInfo &&
      this.props.modalInfo.markerInfo !== newProps.modalInfo.markerInfo
    ) {
      this.setState({
        id: newProps.modalInfo.markerInfo.id,
        name: newProps.modalInfo.markerInfo.name,
        lat: newProps.modalInfo.markerInfo.latitude,
        long: newProps.modalInfo.markerInfo.longitude
      });
    }
  }

  onLocationChange(event) {
    event.preventDefault();
    this.setState({ name: event.target.value });
  }

  onLatChange(event) {
    event.preventDefault();
    this.setState({ lat: event.target.value });
  }

  onLongChange(event) {
    event.preventDefault();
    this.setState({ long: event.target.value });
  }

  closeModal() {
    this.setState({ name: "", lat: null, long: null, id: null });
    this.props.dispatch(
      displayCreateEditModal({ display: false, markerInfo: null })
    );
  }

  saveMarker() {
    if (!(this.state.name && this.state.lat && this.state.long)) {
      toast.error("All fields are required");
      return;
    }
    if ((this.state.lat > 85 || this.state.lat < -85) || (this.state.long > 180 || this.state.long < -180)) {
      toast.error("Valid Latitude range [-85, 85] and Valid Longitude range [-180, 180]");
      return;
    }
    const obj = {
      name: this.state.name,
      lat: Number(this.state.lat),
      long: Number(this.state.long)
    };
    if (this.state.id) {
      obj.id = this.state.id;
      this.props.dispatch(updateMarkers(obj));
    } else {
      this.props.dispatch(saveMarkers(obj));
    }
  }

  render() {
    return (
      <div>
        <Modal
          isOpen={this.props.modalInfo.display}
          onRequestClose={this.closeModal}
          style={{ content: { left: "32%", right: "32%" } }}
        >
          <div>
            <h2>Marker</h2>
          </div>
          <div className="card card-body mg-btm-5">
            <div className="form-group row">
              <label className="w85">Location: </label>
              <input
                className=""
                type="text"
                value={this.state.name}
                onChange={this.onLocationChange.bind(this)}
              />
            </div>
            <div className="form-group row">
              <label className="w85">Latitude: </label>
              <input
                className=""
                type="number"
                value={this.state.lat}
                onChange={this.onLatChange.bind(this)}
              />
            </div>
            <div className="form-group row">
              <label className="w85">Longitude: </label>
              <input
                className=""
                type="number"
                value={this.state.long}
                onChange={this.onLongChange.bind(this)}
              />
            </div>
          </div>
          <div className="btn-div">
            <button
              className="button-cls ml0"
              onClick={this.saveMarker.bind(this)}
            >
              Save
            </button>
            <button className="button-cls" onClick={this.closeModal.bind(this)}>
              Close
            </button>
          </div>
        </Modal>
      </div>
    );
  }
}
