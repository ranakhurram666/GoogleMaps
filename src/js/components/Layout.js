import React from "react";
import { connect } from "react-redux";
import { ToastContainer } from "react-toastify";
// import 'react-toastify/dist/ReactToastify.css';

import { GoogleMap } from "./maps";
import { CreateEditModal, MarkerList } from "./index";
import { getAllMarkers, displayCreateEditModal } from "../actions";

@connect(store => {
  return {
    markers: store.marker.markers,
    modalInfo: store.marker.modalInfo
  };
})
export default class Layout extends React.Component {
  constructor() {
    super();
    this.state = {
      displayModal: false
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllMarkers());
  }

  componentDidMount() {}

  closeCreateEditModal() {
    this.setState({ displayModal: false });
  }

  openCreateEditModal(e) {
    e.preventDefault();
    this.props.dispatch(
      displayCreateEditModal({ display: true, markerInfo: {} })
    );
  }

  render() {
    return (
      <div>
        <GoogleMap markers={this.props.markers}></GoogleMap>
        <br />
        <button
          className="button-cls"
          onClick={this.openCreateEditModal.bind(this)}
        >
          Add Marker
        </button>
        <MarkerList />
        <CreateEditModal
          modalInfo={this.props.modalInfo}
          closeModal={this.closeCreateEditModal.bind(this)}
        ></CreateEditModal>
        <ToastContainer />
      </div>
    );
  }
}
