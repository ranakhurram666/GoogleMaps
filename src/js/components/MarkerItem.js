import React from "react";
import { connect } from "react-redux";
import { displayCreateEditModal, deleteMarker } from "../actions";

@connect(store => {
  return {};
})
export default class MarkerItem extends React.Component {
  editMarker(e) {
    e.preventDefault();
    this.props.dispatch(
      displayCreateEditModal({ display: true, markerInfo: this.props.marker })
    );
  }

  deleteMarker(e) {
    e.preventDefault();
    this.props.dispatch(deleteMarker(this.props.marker.id));
  }

  render() {
    return (
      <div className="card card-body display-inline">
        <h4>{this.props.marker.name}</h4>
        <div className="">
          <label>
            <b>Latitude: </b>
          </label>
          <label>{this.props.marker.latitude}</label>
        </div>
        <div className="">
          <label>
            <b>Longitude: </b>
          </label>
          <label>{this.props.marker.longitude}</label>
        </div>
        <button className="button-cls ml0" onClick={this.editMarker.bind(this)}>
          Edit
        </button>
        <button className="button-cls" onClick={this.deleteMarker.bind(this)}>
          Delete
        </button>
      </div>
    );
  }
}
