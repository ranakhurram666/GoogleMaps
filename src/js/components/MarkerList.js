import React from "react";
import { connect } from "react-redux";
import {MarkerItem} from './index';

@connect((store) => {
    return {
      markers: store.marker.markers
    };
  })
export default class MarkerList extends React.Component {
    constructor() {
        super();
    }
    render() {
        return(
            <div className="card-body">
                {this.props.markers.map((marker) => {
                    return <MarkerItem marker={marker}/>   
                })
                }
            </div>
        )
    }
}