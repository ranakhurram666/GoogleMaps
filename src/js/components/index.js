export { default as Layout } from "./Layout";
export { default as CreateEditModal } from "./CreateEditModal";
export { default as MarkerItem } from "./MarkerItem";
export { default as MarkerList } from "./MarkerList";
