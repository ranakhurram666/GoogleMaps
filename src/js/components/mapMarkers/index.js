export {default as GoogleMapMarker} from './GoogleMapMarker';