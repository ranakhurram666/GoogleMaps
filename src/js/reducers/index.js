import { combineReducers } from "redux";
import marker from './markerReducer';

export default combineReducers({
    marker
});
