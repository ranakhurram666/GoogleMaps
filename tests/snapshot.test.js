import React from "react";
import { MarkerItem, MarkerList, Layout } from "../src/js/components";
import renderer from "react-test-renderer";
import store from "../src/js/store";
import { Provider } from "react-redux";

test("Test MarkerItem component", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MarkerItem
        marker={{ id: 1, name: "Berlin", latitude: 50.393, longitude: 12.272 }}
      ></MarkerItem>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Test MarkerList component", () => {
  const markers = [
    { id: 1, name: "Munich", latitude: 55.393, longitude: 10.272 },
    { id: 3, name: "Berlin", latitude: 53.393, longitude: 18.342 }
  ];
  const component = renderer.create(
    <Provider store={store}>
      <MarkerList markers={markers}></MarkerList>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Test Layout component", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Layout></Layout>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
