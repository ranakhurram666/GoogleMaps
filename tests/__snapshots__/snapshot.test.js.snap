// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout component 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      style={
        Object {
          "height": "100%",
          "overflow": "visible",
          "width": "auto",
        }
      }
    />
  </div>
  <br />
  <button
    className="button-cls"
    onClick={[Function]}
  >
    Add Marker
  </button>
  <div
    className="card-body"
  />
  <div />
  <div
    className="Toastify"
  />
</div>
`;

exports[`Test MarkerItem component 1`] = `
<div
  className="card card-body display-inline"
>
  <h4>
    Berlin
  </h4>
  <div
    className=""
  >
    <label>
      <b>
        Latitude: 
      </b>
    </label>
    <label>
      50.393
    </label>
  </div>
  <div
    className=""
  >
    <label>
      <b>
        Longitude: 
      </b>
    </label>
    <label>
      12.272
    </label>
  </div>
  <button
    className="button-cls ml0"
    onClick={[Function]}
  >
    Edit
  </button>
  <button
    className="button-cls"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;

exports[`Test MarkerList component 1`] = `
<div
  className="card-body"
/>
`;
